File: ios/RNTAddToSiriButtonManager.m
//
//  RNTAddToSiriButtonManager.m
//  RNSiriShortcuts
//
//  Created by Gustavo Parreira on 24/10/2018.
//  Copyright © 2018 Facebook. All rights reserved.
//

#import "React/RCTViewManager.h"
#import "React/RCTBridgeModule.h"
#import "React/RCTEventEmitter.h"

@interface RCT_EXTERN_MODULE(RNTAddToSiriButtonManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(buttonStyle, NSNumber)
RCT_EXPORT_VIEW_PROPERTY(onPress, RCTBubblingEventBlock)
RCT_EXPORT_VIEW_PROPERTY(shortcut, NSDictionary)

@end

File: ios/RNTAddToSiriButtonManager.swift
//
//  RNTAddToSiriButton.swift
//  RNSiriShortcuts
//
//  Created by Gustavo Parreira on 23/10/2018.
//  Copyright © 2018 Facebook. All rights reserved.
//

import Foundation

@available(iOS 12.0, *)
@objc (RNTAddToSiriButtonManager)
open class RNTAddToSiriButtonManager : RCTViewManager {
    @objc override open func view() -> UIView! {
        return SiriButtonView(frame: UIScreen().bounds)
    }
    
    override static public func requiresMainQueueSetup() -> Bool {
        return true
    }
}
